homeassistant/components/sensor/enphase_envoy.py

```diff
@@ -0,0 +1,133 @@
+
+"""
+import logging
+import json
+
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_IP_ADDRESS = 'ip'
+CONF_MONITORED_CONDITIONS = 'monitored_conditions'
+
+DEFAULT_NAMES = [
+    "Envoy Current Energy Production",
+    "Envoy Today's Energy Production",
+    "Envoy Last Seven Days Energy Production",
+    "Envoy Lifetime Energy Production",
+    "Envoy Current Energy Consumption",
+    "Envoy Today's Energy Consumption",
+    "Envoy Last Seven Days Energy Consumption",
+    "Envoy Lifetime Energy Consumption"]
+
+SENSOR_TYPES = [
+    "production", "daily_production", "7_days_production",
+    "lifetime_production", "consumption", "daily_consumption",
+    "7_days_consumption", "lifetime_consumption"]
+
+ICON = 'mdi:flash'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_IP_ADDRESS): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS): vol.All(cv.ensure_list)
+
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Enphase Envoy sensor."""
+    ip_address = config.get(CONF_IP_ADDRESS)
+    monitored_conditions = config.get(CONF_MONITORED_CONDITIONS, {})
+
+
+# Iterate through the list of sensors, adding it if either monitored
+# conditions has been left out of the config, or that the given sensor
+# is in the monitored conditions list
+    for i in range(8):
+        if monitored_conditions == {} or \
+            SENSOR_TYPES[i] in monitored_conditions:
+                add_devices([Envoy(ip_address, DEFAULT_NAMES[i],
+                            SENSOR_TYPES[i])], True)
+
+
+class Envoy(Entity):
+    """Implementation of the Enphase Envoy sensors."""
+    def __init__(self, ip_address, name, type):
+        """Initialize the sensor."""
+        self._url = "http://{}/production.json".format(ip_address)
+        self._name = name
+
+        if type == 'production' or type == 'consumption':
+            self._unit_of_measurement = 'watts'
+        else:
+            self._unit_of_measurement = "watt hours"
+
+        self._type = type
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ICON
+
+    def update(self):
+        """Get the energy production data from the Enphase Envoy."""
+        import requests
+
+        try:
+            response = requests.get(self._url, timeout=5)
```

Comment:
Why not use the RestData class (components/sensor/rest.py) here? May save a bit of code duplication.